core/src/main/scala/com/carto/analyticstoolbox/spark/spatial/OptimizeSpatial.scala

```diff
@@ -42,23 +43,27 @@ object OptimizeSpatial extends Serializable {
     // overwrite the output table
     try ssc.sql(s"DROP TABLE $outputTable;") catch { case _: AnalysisException => // e.printStackTrace() }*/
 
+    // Decide, based on column type of geometry, which parsing function to use
+    val table_df = ssc.table(sourceTable)
+    val parseGeom =
+      if (table_df.schema(geomColumn).dataType == "binary") "ST_geomFromTWKB"
+      else "ST_geomFromWKT"
+
     // view creation
     // SQL definition is easier and more readable
     ssc.sql(
       s"""
          |CREATE TEMPORARY VIEW ${sourceTable}_idx_view AS(
          |  WITH orig_q AS (
          |    SELECT
-         |      * EXCEPT(geom),
-         |      geom AS wkt,
-         |      ST_geomFromWKT(geom) AS geom
+         |      * EXCEPT($geomColumn),
+         |      $geomColumn AS wkt,
+         |      $parseGeom($geomColumn) AS geom
          |      FROM $sourceTable
          |    )
          |    SELECT
          |      *,
-         |      st_z2LatLon(geom) AS z2,
-         |      st_extentFromGeom(geom) AS bbox,
-         |      st_partitionCentroid(geom, $zoom) AS partitioning
```

Comment:
I think I had to clarify the original idea here:

* Remove all unused columns from the *result table*

The result table is the parquet table, we need some columns to partition the table, but we don't need them to query data.

All these columns are used to shape a proper table partitioing. That's how it should look like after this PR:

```scala
""" CREATE TEMPORARY VIEW ${sourceTable}_idx_view AS(
         |  WITH orig_q AS (
         |    SELECT
         |      * EXCEPT($geomColumn),
         |      $parseGeom($geomColumn) AS geom
         |      FROM $sourceTable
         |    )
         |    SELECT
         |      *,
         |      st_z2LatLon(geom) AS __carto_z2,
         |      st_extentFromGeom(geom) AS __carto_index,
         |      st_partitionCentroid(geom, $zoom) AS __carto_partitioning
         |      FROM orig_q
         |      DISTRIBUTE BY __carto_partitioning SORT BY __carto_z2.min, __carto_z2.max
         |  );
```

These lines (https://github.com/CartoDB/analytics-toolbox-databricks/blob/main/core/src/main/scala/com/carto/analyticstoolbox/spark/spatial/OptimizeSpatial.scala#L80-L86) should be changed to: 

```scala
    ssc.sql(
      s"""
         |CREATE TABLE $outputTable
         |USING PARQUET LOCATION '$outputLocation/$outputTable'
         |AS (SELECT * EXCEPT (__carto_partitioning,  __carto_z2) FROM ${sourceTable}_idx_view);
         |""".stripMargin
    )
```